nightfall-optimist/src/routes/proposer.mjs

```diff
@@ -91,20 +95,35 @@ router.post('/register', auth, async (req, res, next) => {
     const isRegistered = proposerAddresses.includes(ethAddress);
 
     let txDataToSign = '';
-    let receipt;
+    let signedTx;
     if (!isRegistered) {
       logger.debug('Register new proposer...');
       txDataToSign = await proposersContractInstance.methods.registerProposer(url, fee).encodeABI();
-      const tx = {
-        from: ethAddress,
-        to: proposersContractInstance.options.address,
-        data: txDataToSign,
-        value: stake,
-        gas: 8000000,
-      };
-      const signedTx = await web3.eth.accounts.signTransaction(tx, ethPrivateKey);
-      receipt = await web3.eth.sendSignedTransaction(signedTx.rawTransaction);
-      logger.debug(`Transaction receipt ${receipt}`);
+
+      // Sign tx
+      const proposersContractAddress = proposersContractInstance.options.address;
+      signedTx = await createSignedTransaction(
+        nonce,
+        ethPrivateKey,
+        ethAddress,
+        proposersContractAddress,
+        txDataToSign,
+        stake,
+      );
+
+      // Submit tx and update db if tx is successful
+      // CHECK - I think the code beyond L132 can be removed (?) then db op could be moved here
+      txsQueue.push(async () => {
```

Comment:
This piece of code is repeated many times, changing only the log message sometimes and the calls inside. Maybe we can think in something to reuse it and make it less verbose. ;)

Reply:
:eyes: That is true, but I would appreciate if I could do this in another PR :pray:

Reply:
Sure! :D